contrib/platform/src/com/sun/jna/platform/win32/COM/util/ProxyObject.java

```diff
@@ -611,6 +611,36 @@ protected HRESULT oleMethod(final int nType, final VARIANT.ByReference pvResult,
 			dp.cNamedArgs = new UINT(_argsLen);
 			dp.rgdispidNamedArgs = new DISPIDByReference(OaIdl.DISPID_PROPERTYPUT);
 		}
+                
+                // Apply "fix" according to
+                // https://www.delphitools.info/2013/04/30/gaining-visual-basic-ole-super-powers/
+                // https://msdn.microsoft.com/en-us/library/windows/desktop/ms221486(v=vs.85).aspx
+                //
+                // Summary: there are methods in the word typelibrary that require both
+                // PROPERTYGET _and_ METHOD to be set. With only one of these set the call
+                // fails.
+                //
+                // The article from delphitools argues, that automation compatible libraries
+                // need to be compatible with VisualBasic which does not distingish methods
+                // and property getters and will set both flags always.
+                //
+                // The MSDN article advises this behaviour: "[...] Some languages cannot 
+                // distinguish between retrieving a property and calling a method. In this 
+                //case, you should set the flags DISPATCH_PROPERTYGET and DISPATCH_METHOD.
+                // [...]"))
+                //
+                // This was found when trying to bind InchesToPoints from the _Application 
+                // dispatch interface of the MS Word 15 type library
+                //
+                // The signature according the ITypeLib Viewer (OLE/COM Object Viewer):
+                // [id(0x00000172), helpcontext(0x09700172)]
+                // single InchesToPoints([in] single Inches);
+                final int finalNType;

```

Comment:
This won't work - in ProxyObject the invocation is dispatched into a separate thread and for that the invoke is wrapped into a Callable, that won't capture ntype if it is not final and if it is final, it can't be changed. so one more variable is needed.

The previous comparisons in the code don't modify ntype.

It would be possible to apply the requested change in COMBindingBaseObject, but that code is identical to the code in ProxyObject and intentionally so.